### DISCUSSION TITLE
How can I setup a node-to-node SSL connection?

### DISCUSSION THREAD

**Discussion by lamhktommy** [citusdata/citus#7531] [Q&A] [answered]
My current setup would be 1 master node and 2 worker nodes. SSL setting in postgresql.conf is like this
![image](https://github.com/citusdata/citus/assets/35717697/899d8678-b699-43a7-a067-453fd6c67d13)

**Comment by JelteF** [comment, marked as answer]
You can use `citus.node_conninfo` to configure the `sslmode` and `sslrootcert` that is used during connections between nodes.

**Reply by lamhktommy** [reply]
Is it by table or by node setting? I am referring to https://docs.citusdata.com/en/v10.2/develop/api_guc.html?highlight=citus.node_conninfo#citus-node-conninfo-text

**Reply by JelteF** [reply]
If you configure it on node A, it will use those settings to connect to both node B and node C for all connections (thus all tables).

If you need different settings when connecting to node B or node C you can use the `pg_dist_authinfo` table. But that shouldn't be necessary for most setups, generally this is only the case when the root certificate is different for each of your nodes (which is not a recommended way of setting up TLS).

**Reply by lamhktommy** [reply]
Is there a quicker way for me to set all tables in a node to have the same SSL cert? I have 2-3 databases in my Postgresql Instance, and each would contain around 20 tables.

**Reply by JelteF** [reply]
SSL cert is always configured at the node level, not at the table level.

**Reply by lamhktommy** [reply]
ohhh sorry, I misread the example in above link. Thanks for the detailed explanation. Have a nice day !

**Reply by lamhktommy** [reply]
but when i run the command it shows this error: 
``` parameter "citus.node_conninfo" cannot be changed now ```

**Reply by JelteF** [reply]
You should configure it in the postgresql.conf file

**Reply by lamhktommy** [reply]
I am not sure how to set citus.node_conninfo in postgres.conf
Any example or reference I can look into?

**Reply by JelteF** [reply]
add a line to the postgresql.conf file like:
```
citus.node_conn = 'sslmode=verify-full sslrootcert=/some/path/to/ca.crt'
```
